frontend/src/components/moderation/ReviewRow.tsx

```diff
@@ -0,0 +1,209 @@
+import { Dialog, Transition } from "@headlessui/react"
+import { useTranslation } from "next-i18next"
+import { Fragment, FunctionComponent, ReactElement, useState } from "react"
+import { submitReview } from "src/asyncs/moderation"
+import { getIntlLocale, getLocale } from "src/localize"
+import { ModerationRequest } from "src/types/Moderation"
+import Button from "../Button"
+import InlineError from "../InlineError"
+import Spinner from "../Spinner"
+import { formatDistance } from "date-fns"
+import { useUserContext } from "src/context/user-info"
+import Link from "next/link"
+
+interface Props {
+  title: string
+  request: ModerationRequest
+}
+
+const ReviewRow: FunctionComponent<Props> = ({ title, request, children }) => {
+  const { t, i18n } = useTranslation()
+  const user = useUserContext()
+
+  const [modalState, setModalState] = useState<"approve" | "reject">("approve")
+  const [modalVisible, setModalVisible] = useState(false)
+
+  const [comment, setComment] = useState<string | undefined>(undefined)
+
+  const modalTitle =
+    modalState === "reject" ? t("moderation-reject") : t("moderation-approve")
+
+  const confirmText =
+    modalState === "reject"
+      ? t("moderation-reject")
+      : comment
+      ? t("moderation-approve-with-comment")
+      : t("moderation-approve")
+
+  const [status, setStatus] = useState<
+    "idle" | "pending" | "error" | "success"
+  >("idle")
+  const [error, setError] = useState<string>()
+
+  const submit = async (approve: boolean) => {
+    setStatus("pending")
+    try {
+      await submitReview(request.id, approve, comment)
+      setStatus("success")
```

Comment:
I think we might want to introduce some kind of toasts, to give the user a feedback in one of the next iterations.

Reply:
I just remembered, that we already have toasts :)

Reply:
It currently gives the error inline:

![image](https://github.com/flathub/website/assets/24497353/4ebcf0d0-b3b0-4624-a4cd-f141c6b1c3cb)